pinot-core/src/test/java/org/apache/pinot/queries/ArgMinMaxTest.java

```diff
@@ -0,0 +1,613 @@
+    assertEquals(rows.get(1)[0], 999L);
+    assertEquals(rows.get(0)[1], 999.5F);
+    assertEquals(rows.get(1)[1], 999.5F);
+    assertEquals(rows.get(0)[2], 1999D);
+    assertEquals(rows.get(1)[2], 1999D);
+    assertEquals(rows.get(0)[3], new Integer[]{0, 1, 2});
+    assertEquals(rows.get(1)[3], new Integer[]{0, 1, 2});
+    assertEquals(rows.get(0)[4], new String[]{"a0", "a01", "a02"});
+    assertEquals(rows.get(1)[4], new String[]{"a0", "a01", "a02"});
+    assertEquals(rows.get(0)[5], 0);
+    assertEquals(rows.get(1)[5], 0);
+    assertEquals(rows.get(0)[6], "360000");
+    assertEquals(rows.get(1)[6], "360000");
+    assertEquals(rows.get(0)[7], 600D);
+    assertEquals(rows.get(1)[7], 600D);
+    assertEquals(rows.get(0)[8], 1683138373879L - 1999L);
+    assertEquals(rows.get(1)[8], 1683138373879L - 1999L);
+    assertEquals(rows.get(0)[9], new Double[]{600D, 600D * 600D, 600D * 600D * 600D});
+    assertEquals(rows.get(1)[9], new Double[]{600D, 600D * 600D, 600D * 600D * 600D});
+    assertEquals(rows.get(0)[10], "{\"name\":\"John\",\"age\":600,\"car\":null}");
+    assertEquals(rows.get(1)[10], "{\"name\":\"John\",\"age\":600,\"car\":null}");
+
+    // Inter segment data type test for boolean column
+    query = "SELECT arg_max(booleanColumn, booleanColumn) FROM testTable";
+
+    brokerResponse = getBrokerResponse(query);
+    resultTable = brokerResponse.getResultTable();
+    rows = resultTable.getRows();
+
+    assertEquals(rows.size(), 2000);
+    for (int i = 0; i < 2000; i++) {
+      assertEquals(rows.get(i)[0], 1);
+    }
+
+    // Inter segment mix aggregation function with different result length
+    // Inter segment string column comparison test, with dedupe
+    query = "SELECT sum(intColumn), argmin(stringColumn, doubleColumn), argmin(stringColumn, stringColumn), "
+        + "argmin(stringColumn, doubleColumn, doubleColumn) FROM testTable";
+
+    brokerResponse = getBrokerResponse(query);
+    resultTable = brokerResponse.getResultTable();
+    rows = resultTable.getRows();
+
+    assertEquals(rows.size(), 4);
+
+    assertEquals(rows.get(0)[0], 7996000D);
+    assertEquals(rows.get(0)[1], 8D);
+    assertEquals(rows.get(0)[2], "a11");
+    assertEquals(rows.get(0)[3], 8D);
+
+    assertNull(rows.get(1)[0]);
+    assertEquals(rows.get(1)[1], 18D);
+    assertEquals(rows.get(1)[2], "a11");
+    assertEquals(rows.get(1)[3], 8D);
+
+    assertNull(rows.get(2)[0]);
+    assertEquals(rows.get(2)[1], 8D);
+    assertEquals(rows.get(2)[2], "a11");
+    assertNull(rows.get(2)[3]);
+
+    assertNull(rows.get(3)[0]);
+    assertEquals(rows.get(3)[1], 18D);
+    assertEquals(rows.get(3)[2], "a11");
+    assertNull(rows.get(3)[3]);
+
+    // Test transformation function inside argmax/argmin, for both projection and measuring
+    // the max of 3000x-x^2 is 2250000, which is the max of 3000x-x^2
+    query = "SELECT sum(intColumn), argmax(3000 * doubleColumn - intColumn * intColumn, doubleColumn),"
```

Comment:
So essentially the first set of arguments for lex ordering (for min or max) can be a mix of identifier (column) or a transform (scalar or non-scalar). Correct ?

Is the same true for `projectionColumn` ? Can we project the transform instead of identifier ?

Reply:
yes there are tests for projection of transformed cols in the testcase

Reply:
```
    query = "SELECT sum(intColumn), argmax(3000 * doubleColumn - intColumn * intColumn, doubleColumn),"
        + "argmax(3000 * doubleColumn - intColumn * intColumn, 3000 * doubleColumn - intColumn * intColumn),"
        + "argmax(3000 * doubleColumn - intColumn * intColumn, doubleColumn), "
        + "argmin(replace(stringColumn, \'a\', \'bb\'), replace(stringColumn, \'a\', \'bb\'))"
        + "FROM testTable";
```